### PR TITLE
chore(ci): Add go cache

### PR DESCRIPTION
**Pull request by yevgenypats**

### CODE REVIEW

**Comment by yevgenypats** [comment]
either doing something wrong or this is bug - https://github.com/actions/setup-go/issues/256 as can't find the path to `go.sum` and `go.mod`

File: .github/workflows/source_gcp.yml [hunk 1]
```diff
@@ -21,7 +21,9 @@ jobs:
       - name: Set up Go 1.x
         uses: actions/setup-go@v3
         with:
-          go-version: 1.19.1
+          go-version-file: ./plugins/source/gcp/go.mod
+          cache: true
+          cache-dependency-path: ./plugins/source/gcp/go.sum
```
Comment (erezrokah) [inline .github/workflows/source_gcp.yml:26]:
```suggestion
          go-version-file: plugins/source/gcp/go.mod
          cache: true
          cache-dependency-path: plugins/source/gcp/go.sum
```

**Comment by erezrokah** [comment]
>either doing something wrong or this is bug - https://github.com/actions/setup-go/issues/256 as can't find the path to go.sum and go.mod

You first need to checkout the code, then do the `setup-go`

**Comment by yevgenypats** [comment]
> > either doing something wrong or this is bug - [actions/setup-go#256](https://github.com/actions/setup-go/issues/256) as can't find the path to go.sum and go.mod
> 
> You first need to checkout the code, then do the `setup-go`

lol. good catch. I knew I had to go to sleep :)

**Pull request merged**